packages/odyssey-react-mui/src/labs/DataTable.tsx

```diff
@@ -480,28 +501,139 @@ const DataTable = ({
+
+                if (nextRowFromDataAfterDownArrow) {
+                  getRowFromTableAndSetHovered(
+                    table,
+                    nextRowFromDataAfterDownArrow.id
+                  );
+                }
+              }
 
-        setDraggingRow(null);
-      },
+              if (isArrowUp) {
+                const nextRowFromDataAfterUpArrow = data[index - 1];
 
-      onDragCapture: () => {
-        if (!draggingRow && table.getState().draggingRow?.id) {
-          setDraggingRow(table.getState().draggingRow);
-        }
-      },
+                if (nextRowFromDataAfterUpArrow) {
+                  getRowFromTableAndSetHovered(
+                    table,
+                    nextRowFromDataAfterUpArrow.id
+                  );
+                }
+              }
+            } else {
+              if (isArrowDown || isArrowUp) {
+                if (isArrowDown) {
+                  const shouldDisableReOrder =
+                    currentIndex === page * resultsPerPage - 1;
+                  if (shouldDisableReOrder) {
+                    return;
+                  }
+                }
+
+                if (isArrowUp) {
+                  const shouldDisableReOrder =
+                    currentIndex === 0 ||
+                    currentIndex === (page - 1) * resultsPerPage;
+                  if (shouldDisableReOrder) {
+                    return;
+                  }
+                }
+
+                const nextIndex = isArrowDown ? row.index + 1 : row.index - 1;
+                const nextRowFromData = data[nextIndex];
+
+                if (nextRowFromData) {
+                  getRowFromTableAndSetHovered(table, nextRowFromData.id);
+                }
+              }
+            }
+          } else {
+            if (isSpaceOrEnter) {
+              setDraggingRow(row);
+            }
+          }
+        },
+        onBlur: () => {
+          resetDraggingAndHoveredRow(table);
+        },
+        onDragEnd: () => {
+          const cols = table.getAllColumns();
+          cols[0].toggleVisibility();
+
+          const { draggingRow, hoveredRow } = table.getState();
+          if (draggingRow) {
+            handleReordering({
+              rowId: draggingRow.id,
+              newIndex: (hoveredRow as MRT_RowData).index,
+            });
+          }
+
+          setDraggingRow(null);
+        },
+
+        onDragCapture: () => {
+          if (!draggingRow && table.getState().draggingRow?.id) {
+            setDraggingRow(table.getState().draggingRow);
+          }
+        },
+        sx: {
+          padding: odysseyTokens.Spacing1,
+          borderRadius: odysseyTokens.BorderRadiusMain,
+
+          "&:focus-visible": {
+            boxShadow: `0 0 0 2px ${odysseyTokens.HueNeutralWhite}, 0 0 0 4px ${odysseyTokens.PalettePrimaryMain}`,
+            outline: "2px solid transparent",
+            outlineOffset: "1px",
+          },
+        },
+      };
```

Comment:
Consider extracting this out to separate function for 2 reasons:
- This function is being recreated each time  the parent is created
- it would be much easier to read as `onDragCapture: onDragCaptureHandler`